src/Http/Http.Extensions/src/DefaultProblemDetailsWriter.cs

```diff
@@ -34,13 +34,13 @@ public bool CanWrite(ProblemDetailsContext context)
         {
             return true;
         }
-
+    
         for (var i = 0; i < acceptHeader.Count; i++)
         {
             var acceptHeaderValue = acceptHeader[i];
-
-            if (_jsonMediaType.IsSubsetOf(acceptHeaderValue) ||
-                _problemDetailsJsonMediaType.IsSubsetOf(acceptHeaderValue))
+            //One of the media types needs to be checked if it's a subset of the accepted header value
+            //Application/json is a subset of */* but */* is not a subset of application/json
+            if(acceptHeaderValue.IsSubsetOf(_jsonMediaType) || acceptHeaderValue.IsSubsetOf(_problemDetailsJsonMediaType) || _jsonMediaType.IsSubsetOf(acceptHeaderValue))
```

Comment:
Is there a reason you left the third clause (_jsonMediaType.IsSubsetOf(acceptHeaderValue)) in this check?